### PR TITLE
impl switch op forward part

### PR DESCRIPTION
**Pull request by jacquesqiao**
fix: https://github.com/PaddlePaddle/Paddle/issues/8142
project: https://github.com/PaddlePaddle/Paddle/issues/7769

### CODE REVIEW

File: paddle/framework/framework.proto [hunk 1]
```diff
@@ -46,6 +47,7 @@ message OpDesc {
     repeated bool bools = 11;
     optional int32 block_idx = 12;
     optional int64 l = 13;
+    repeated int32 block_idxs = 14;
```
Comment (Superjomn) [inline paddle/framework/framework.proto]: what's the relation between `block_idx` and `block_idxs`?

can they merge into one field?  given `optional = repeated[size=0+]`

Comment (jacquesqiao) [reply paddle/framework/framework.proto]: block_idxs = vector<block_idx>

Comment (jacquesqiao) [reply paddle/framework/framework.proto]: I will take a look if they can be merged

**Comment by jacquesqiao** [comment]
will implement in https://github.com/PaddlePaddle/Paddle/pull/8184

**Pull request closed without merging**